pkgs/development/libraries/unixODBCDrivers/default.nix

```diff
@@ -181,6 +181,104 @@
     };
   };
 
+  msodbcsql18 = stdenv.mkDerivation rec {
+      pname = "msodbcsql${versionMajor}";
+      version = "${versionMajor}.${versionMinor}.${versionAdditional}${versionSuffix}";
+
+      versionMajor = "18";
+      versionMinor = "1";
+      versionAdditional = "1.1";
+      versionSuffix = lib.strings.optionalString stdenv.targetPlatform.isLinux "-1";
+
+    src =
+      if stdenv.targetPlatform.isDarwin
+      then
+        if stdenv.targetPlatform.isAarch64
+        then
+          fetchurl {
+            url = "https://download.microsoft.com/download/6/4/0/64006503-51e3-44f0-a6cd-a9b757d0d61b/${pname}-${version}-arm64.tar.gz";
+            sha256 = "116xl8r2apr5b48jnq6myj9fwqs88yccw5176yfyzh4534fznj5x";
+          }
+        else
+          fetchurl {
+            url = "https://download.microsoft.com/download/6/4/0/64006503-51e3-44f0-a6cd-a9b757d0d61b/${pname}-${version}-amd64.tar.gz";
+            sha256 = "1fn80byn1yihflznxcm9cpj42mpllnz54apnk9n46vzm2ng2lj6d";
+          }
+      else
+        if stdenv.targetPlatform.isAarch64
+        then
+          fetchurl {
+            url = "https://packages.microsoft.com/debian/11/prod/pool/main/m/${pname}/${pname}_${version}_arm64.deb";
+            sha256 = "0zphnbvkqdbkcv6lvv63p7pyl68h5bs2dy6vv44wm6bi89svms4a";
+          }
+        else
+          fetchurl {
+            url = "https://packages.microsoft.com/debian/11/prod/pool/main/m/${pname}/${pname}_${version}_amd64.deb";
+            sha256 = "1f0rmh1aynf1sqmjclbsyh2wz5jby0fixrwz71zp6impxpwvil52";
+          };
+
+    nativeBuildInputs =
+      if stdenv.targetPlatform.isDarwin
+      then
+        [
+          # Fix up the names encoded into the dylib, and make them absolute.
+          fixDarwinDylibNames
+        ]
+      else
+        [
+          dpkg
+          patchelf
+        ];
+
+      unpackPhase = lib.strings.optionalString stdenv.targetPlatform.isLinux "dpkg -x $src ./";
+
+      installPhase =
```

Comment:
I would prefer to keep it as-is; I find staggered conditions like this, where something inside the block is indented less than the thing outside the block, quite hard to read.

Reply:
> I would prefer to keep it as-is; I find staggered conditions like this, where something inside the block is indented less than the thing outside the block, quite hard to read.

Within reasonably common & sensible coding style and coherence, it is your right to use whatever indentation and coding style you choose for packages, if you are consistent. The advantage of the indentation I suggested, is that it minimizes diffs in case lines from one platform are moved to other platforms / conditions etc.

Reply:
A tough one: one's easier when reading code, one's easier when reading a diff.

I will keep it as it is but I totally understand why you have a different preference.